### DISCUSSION TITLE
Shoutrrr not working: 401 Unauthorized

### DISCUSSION THREAD

**Discussion by ghost** [qdm12/ddns-updater#242] [Q&A] [answered]
Hello everyone,

I have been running the ddns-updater for a few days and everything works so far. My problem, however, is that notifications can not be sent via Telegram or Discord. The error "failed to send notification: response status code 401 Unauthorized" occurs every time the container is started and if the IP address is changed successfully.

I configured the variable SHOUTRRR_ADDRESSES as follows:

discord://token@webhookid,telegram://token@telegram?channels=@channelname

I couldn't find a solution to this problem anywhere.
I hope you can help me.

Best regards
Mathilda

**Comment by qdm12** [comment]
Hi there!

1. Do you have any commas in your connection strings perhaps?
2. Do you get the `Launched with N records to watch` message at all?
3. Does it work with only the discord connection string instead of both?
4. Does it work with only the Telegram connection string instead of both?

I've pushed 8b327f81c736f3a421e838b94c34e2bb8a120189 which should add which service is failing instead of just `failed to send notification: response status code 401 Unauthorized`. It should be built in a few minutes, and you can then pull `docker pull qmcgaw/ddns-updater`.

**Reply by ghost** [reply]
Hi there,

1. There is no comma in any of the configurations
2. No, I am not receiving this message. The log setting is on Denug.
3./4. I've also tried Telegram and Discord individually, but without success.

I get the error message for Telegram and Discord. In the opening question I only wrote this in summary.

best regards

**Reply by qdm12** [reply]
So do you get the 401 error twice then? Try with the latest image to check from which service you get that error.

401 usually means that you're unauthorized so your credentials might be wrong. Maybe try adding double quotes around the entire value for the environment variable?

**Reply by ghost** [reply]
Hi there,

I get the error in the same way with both services. At first I only had Telegram, which is why I thought there was a configuration bug here. However, the same error message comes up with Discord.

I have now installed the latest update and have tested the point with the double quotes, but without success.
The webhook as such works for other services without any problems. Only with Shoutrrr do I currently have the problems, because the formatting is different here too. Discord itself lists "The Authorization header was missing or invalid" for code 401.
Does Shoutrrr work for you?

Best regards

**Reply by qdm12** [reply]
I use Shoutrrr with Gotify only and it does work.

I updated the Shoutrrr library to the most recent release (from v0.4.4 to v0.5.1), maybe try pulling the image and try again? I see some Discord related fixes between the two releases.

I'll try with Discord and get back to you in a few hours.

**Reply by qdm12** [reply]
My apologies, it was because the program was lower-casing the value of the environment variable, so changing the tokens as well 😢

Fixed in 01f40444047c81700d3a7cbaee7966c6607dfe0e for the latest image, and also made a release tag `:v2.4.1` with just that bug fix on top of the `v2.4.0` release.

EDIT: Let me know if it works/doesn't work, but it should now! 😉

**Comment by fredericrous** [comment, marked as answer]
I'm getting the 401 error on discord too.
According to the following issue, it can be solved by configuring ssl certificates on the image:

- https://github.com/containrrr/shoutrrr/issues/54
- https://github.com/containrrr/shoutrrr/blob/main/Dockerfile

**Reply by qdm12** [reply]
Hey @fredericrous thanks for your input!

I don't think this can be a cause to the `401` error. If tls can't be established, I doubt it would even allow communication at all (so no status returned). In addition, root certificates are in the binary with:

https://github.com/qdm12/ddns-updater/blob/904c59d6ac55456133d9bdd5fe3213749ceb8a53/cmd/updater/main.go#L18

**Reply by qdm12** [reply]
Fixed in 01f40444047c81700d3a7cbaee7966c6607dfe0e for the latest image, and also made a release tag `:v2.4.1` with just that bug fix on top of the `v2.4.0` release.

**Reply by ghost** [reply]
Hey @qdm12,

I just downloaded the latest image and now everything is working as it should. In addition to Discord and Telegram, e-mail also works. e-mail did not work with the last image either.

Many thanks for the quick support.

Best regards

**Reply by fredericrous** [reply]
I confirm discord is working now 🍾 🍾 🍾 
Thanks qdm12
